### PR TITLE
Ashen tree!

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
Adds a new lava land flora, a burnt tree!
The burnt tree or Ashed Tree can have sap in it to collect honey, useful to gather for the cook or as an ashwalker to get surger/basic toxin healing
image
image
image

Gives a random amount of honey 5-15
Gives a random amount of coal 5-15

You can cut down the ashen tree to get charcoal to gather for a grill or crafting easyer 
## Why It's Good For The Game

Adds a new way to get coal, and honey thats not bee related and thematic to the lava lands below.

## Changelog
:cl:
add: New type of flora to lava land, an Ash Tree, can be harvested for coal or if lucky used to gather honey. Just use a container!
/:cl:

Many thanks to BlueWildrose for aiding in code

### CODE REVIEW

**Comment by timothyteakettle** [comment]
this makes honey too easy to get in large quantities

**Comment by BlueWildrose** [comment]
> For some reason cutting down a tree well it has a bowl or container in it, will delete it. Was not really sure how to correct that...

I recommend looking up closet code as a reference. When closets get deleted, they drop all of their contents onto the turf beneath it. Do the same for the tree.

**Comment by Trilbyspaceclone** [comment]
> this makes honey too easy to get in large quantities

What amount do you want then?

**Comment by KrabSpider** [comment]
> > this makes honey too easy to get in large quantities
> 
> What amount do you want then?

Most definitely not 60 or 40.

**Comment by Solaris-Shade** [comment]
I'm all for this, this looks like it could be FUN!

The easy honey thing could lead to unintended consequences. I'd say keep an eye out if the trees become the go to for getting honey than all other methods. If the case then I'd say drop the honey amount even more.

**Comment by Trilbyspaceclone** [comment]
What numbers do you suggest?

File: code/modules/mining/lavaland/ash_tree.dm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	layer = FLY_LAYER
+	var/coal_amount = 5 //amout of coal in are tree, simular to logs
+	var/sap = FALSE //Do we have sap?
+	var/are_sap = /datum/reagent/consumable/honey //What reagent we have
+	//var/tapping_items = list(/obj/item/reagent_containers/glass) - current dosnt work commiting it out
+	var/harvest_sap_time = 60 //This is in seconds, and now long we wait till are tree is tapped
+	var/container_used
+	var/sap_amount
+
+/obj/structure/flora/ashtree/New()
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: /New() is banned

File: code/modules/mining/lavaland/ash_tree.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	..()
+	if(prob(50))
+		sap = TRUE
+		icon_state = sap_icon_state
+		desc = "A once large tree now burnt like the lands around it. This one seems to have a sap still inside."
+		sap_amount = rand(5,15) //good amout of honey
+	coal_amount = rand(5,15) //We give a random amout
+	SSblackbox.record_feedback("tally", "Honey Tree", 1, "Trees Spawned") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: Er, is this necessary? Hate to nitpick but uh, feedback isn't usually done for procedural generation tracking. If you have a reason for it it's fine.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	var/tabbed_icon_state = "ashtree_maple" //What we look like when tapping
+	name = "ashed tree"
+	desc = "A once large tree now burnt like the lands around it."
+	density = TRUE
+	pixel_x = -16
+	layer = FLY_LAYER
+	var/coal_amount = 5 //amout of coal in are tree, simular to logs
+	var/sap = FALSE //Do we have sap?
+	var/are_sap = /datum/reagent/consumable/honey //What reagent we have
+	//var/tapping_items = list(/obj/item/reagent_containers/glass) - current dosnt work commiting it out
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: Don't leave commented code in on a finished PR.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	var/sap_icon_state = "ashtree_maple" //Are icon when we are full of honey or other sap
+	var/tabbed_icon_state = "ashtree_maple" //What we look like when tapping
+	name = "ashed tree"
+	desc = "A once large tree now burnt like the lands around it."
+	density = TRUE
+	pixel_x = -16
+	layer = FLY_LAYER
+	var/coal_amount = 5 //amout of coal in are tree, simular to logs
+	var/sap = FALSE //Do we have sap?
+	var/are_sap = /datum/reagent/consumable/honey //What reagent we have
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: var/sap_type please. Also, put comments like this

/// What reagent our sap is
var/sap_type = /datum/reagent/consumable/honey

So the code doumentation generator works properly.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 5]
```diff
@@ -0,0 +1,82 @@
+//Houses the ash tree, a lava land tree that has been burning for quite some time making a maple like sweetener.
+
+/obj/structure/flora/ashtree
+	gender = PLURAL //same as other tree
+	icon = 'icons/obj/lavaland/ash_tree.dmi'
+	icon_state = "ashtree"
+	var/sap_icon_state = "ashtree_maple" //Are icon when we are full of honey or other sap
+	var/tabbed_icon_state = "ashtree_maple" //What we look like when tapping
+	name = "ashed tree"
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm:4]: We prefer to have override variables **above** variable defines, it makes the code a bit cleaner.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		sap_amount = rand(5,15) //good amout of honey
+	coal_amount = rand(5,15) //We give a random amout
+	SSblackbox.record_feedback("tally", "Honey Tree", 1, "Trees Spawned") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+//So we dont lose are bowls, stolen form closet code
+/obj/structure/flora/ashtree/Destroy()
+	dump_contents(override = FALSE)
+	return ..()
+
+/obj/structure/flora/ashtree/proc/dump_contents(override = TRUE) //Override is for not revealing the locker electronics when you open the locker, for example
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: What is this for?

File: code/modules/mining/lavaland/ash_tree.dm [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+			icon_state = tabbed_icon_state
+			sap = FALSE
+			container_used = W
+			W.forceMove(src) //So we dont lose are bowl when cutting it down + needed for the harvest sap proc
+			addtimer(CALLBACK(src, .proc/harvest_sap), harvest_sap_time SECONDS)
+		else
+			to_chat(user, "<span class='notice'>There is no sap to collect.</span>")
+
+	if(coal_amount && (!(flags_1 & NODECONSTRUCT_1)))
+		if(W.sharpness && W.force > 0)
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: Early returns, aka if(!W.sharpness || !W.force) return.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+//If you hit it with a sharp force aboe 0 item it chops it down, unlike trees tho it dosnt give wood as its already charcoal
+//Also dosnt have a stump
+/obj/structure/flora/ashtree/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers))
+		if(sap)
+			user.visible_message("<span class='notice'>[user] pokes [src] and places a container under the [W].</span>","<span class='notice'>You set up [src] with [W].</span>")
+			icon_state = tabbed_icon_state
+			sap = FALSE
+			container_used = W
+			W.forceMove(src) //So we dont lose are bowl when cutting it down + needed for the harvest sap proc
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: use a mob's transferItemToLoc, don't just forcemove.

File: code/modules/mining/lavaland/ash_tree.dm [hunk 9]
```diff
@@ -0,0 +1,82 @@
+//Houses the ash tree, a lava land tree that has been burning for quite some time making a maple like sweetener.
+
+/obj/structure/flora/ashtree
+	gender = PLURAL //same as other tree
+	icon = 'icons/obj/lavaland/ash_tree.dmi'
+	icon_state = "ashtree"
```
Comment (silicons) [inline code/modules/mining/lavaland/ash_tree.dm]: Please put these variables beneath name/desc/density, it's really weird to break convention here. Minor nitpick and not a huge deal but yeah.

**Review by silicons — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/ash_tree.dm (hunk 6 above)
Comment (Trilbyspaceclone) [reply code/modules/mining/lavaland/ash_tree.dm]: When cutting down a tree, so we dont null are bowl if one is present, of other item inside it

File: code/modules/mining/lavaland/ash_tree.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/modules/mining/lavaland/ash_tree.dm]: Its to get data sets on how many spawn a round, to then get how many are cut || harvested to get %s

**Comment by Trilbyspaceclone** [comment]
This pr good or?

**Pull request merged**